js_modules/dagster-ui/packages/ui-core/src/partitions/OpJobPartitionsView.tsx

```diff
@@ -287,9 +292,19 @@ export const OpJobPartitionsViewContent = React.memo(
             <Button onClick={() => setShowSteps(!showSteps)} active={showBackfillSetup}>
               {showSteps ? 'Hide per-step status' : 'Show per-step status'}
             </Button>
+            <Button
+              onClick={() => partitionsQueryResult.refetch()}
+              loading={partitionsQueryResult.loading}
+              disabled={partitionsQueryResult.loading}
+            >
+              Refresh
+            </Button>
             {canLaunchPartitionBackfill ? (
               <Button
-                onClick={() => setShowBackfillSetup(!showBackfillSetup)}
+                onClick={() => {
+                  void partitionsQueryResult.refetch();
+                  setShowBackfillSetup(!showBackfillSetup);
```

Comment:
Are we worried about the asyncness here? We might need to wait for the refetch or show some kind of spinner in the modal saying we're refreshing the data and that the data in the modal might be out of date?

Reply:
Edit: added a refresh label here in the top right within the modal that is linked to the same state so it's more obvious:

![Uploading image.png…]()